content/tutorials/build-apps/build-a-browser-wallet-in-js.md

```diff
@@ -0,0 +1,194 @@
+
+6. Now, let's add the code to `index.js` file to fetch the account and ledger details and display them on the home page. Copy the code written below to the `index.js` file. Here we render the wallet details using the function we defined in `get-wallet-details.js`. In order to make sure we have up to date ledger data, we are using the [ledger stream](subscribe.html#ledger-stream) to listen for ledger close events.
+
+{{ include_code("_code-samples/build-a-wallet/js/index.js", language="js") }}
+
+The one other thing we do here is add events to two buttons - one to send XRP and one to view transaction history. They won't work just yet—we'll implement them in the next steps.
+
+Now the application is ready to run. You can start it in dev mode using the following command:
+
+```bash
+yarn dev
+```
+
+Your terminal should output a URL which you can use to open your app in a browser. This dev site automatically updates to reflect any changes you make to the code.
+
+### 3. Create the Send XRP Page
+
+Now that we've created the home page, we can move on to the "Send XRP" page. This is what allows this wallet to manage your account's funds.
+
+![Send XRP Page Screenshot](img/js-wallet-send-xrp.png)
+
+1. Create a folder named `send-xrp` in the `src` directory.
+
+2. Inside the `send-xrp` folder, create two files named `send-xrp.js` and `send-xrp.html`.
+
+3. Copy the contents of the [send-xrp.html]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/src/send-xrp/send-xrp.html) file to your `send-xrp.html` file. The provided HTML code includes three input fields for the destination address, amount, and destination tag, each with their corresponding labels.
+
+4. Now that we have the HTML code, let's add the JavaScript code. In the `helpers` folder, create a new file named `submit-transaction.js` and copy the code written below to the file. In this file, we are using the [submit](submit.html) method to submit the transaction to the XRPL. Before submitting every transaction needs to be signed by a wallet, learn more about [signing](sign.html) a transaction.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/helpers/submit-transaction.js", language="js") }}
+
+5. Now back to the `send-xrp.js` file, copy the code written below to the file. In this piece of code we are first getting all the DOM elements from HTML and adding event listners to update & validate the fields based on the user input. Using `renderAvailableBalance` method we display the current available balance of the wallet. `validateAddress` function validates the user address, and the amount is validated using a regular expression. When all the fields are filled with correct inputs, we call the `submitTransaction` function to submit the transaction to the ledger.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/send-xrp/send-xrp.js", language="js") }}
+
```

Comment:
I'd recommend a small section here showcasing what the app should now be able to do. E.g. A screenshot, and something like "You can now click 'Send XRP' to try creating your own transaction! You can use this example to send XRP to the testnet faucet to try it out"

Testnet faucet account: rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe
Amount: 100 
Destination Tag: (Not usually necessary unless you're paying an account tied to an exchange)

Reply:
Added